inc/class-core-sitemaps-renderer.php

```diff
@@ -91,6 +91,22 @@ public function get_sitemap_index_stylesheet_url() {
 	public function render_index( $sitemaps ) {
 		header( 'Content-type: application/xml; charset=UTF-8' );
 
+		if ( ! class_exists( 'SimpleXMLElement' ) ) {
+			add_filter( 'wp_die_handler', static function() { return '_xml_wp_die_handler'; } );
```

Comment:
It didn't seem to work without this. A regular HTML error was displayed otherwise.

Reply:
Does Google & Co. use and XML Accept header? https://developer.wordpress.org/reference/functions/wp_is_xml_request/

This might be a feature... If you open it in a browser you get the HTML and Search Engines "see" the XML.

Reply:
I'd rather explicitly send XML all the time.

When integrating this into core, we'd add a new function like `is_sitemap()`. The `wp_die()` function already contains `is_feed()` checks next to `wp_is_xml_request()`, so we could just amend this with a `is_sitemap()` check.